### ISSUE TITLE
[DataGrid] singleSelect column sets invalid value when starting edit mode by keypress

### ISSUE THREAD

**Issue by cbn-falias** [mui/mui-x#11221]
### Search keywords

datagrid singleselect keypress

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: https://mui.com/x/react-data-grid/editing/#full-featured-crud

Steps:
1. Set focus on any cell in the "Department" column
2. Press any letter or number key
3. 🔥 Single key gets instantly applied as new value
    * combobox selection is empty
    * when saving the row, the invalid value remains

### Current behavior

The pressed key gets instantly applied as new edited value which is probably invalid in most use cases. It behaves like it's a simple textfield.

### Expected behavior

**Minimal fix**
Simply start the edit mode without any change.

**Optional**
Instantly expand the dropdown and pass the pressed key as search value inside the dropdown.

**Comment by DanailH** [comment]
@cbn-falias I agree that it looks broken and it is inconsistent. I would advise for now to with a custom validation as a workaround while we look into fixing it internally.

https://mui.com/x/react-data-grid/editing/#validation

**Comment by DanailH** [comment]
yes, that is another option. I was approaching it more from the perspective of letting the user that they are about to do something that is not allowed, but your approach also works.
